Migrate to using YAML files to store product information

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request is a re-run of #412 with the issues raised (hopefully) resolved. It converts the existing `SOFTWARE-LIST.md` Markdown file into individual YAML files grouped by vendor. When a `push` is made to the `develop` branch the `update_software_list.yml` GitHub Actions workflow will run and update the comprehensive `data/cisagov.yml` file from the separate source YAML files and then generate an updated `SOFTWARE-LIST.md`.

This pull request is in concert with the documentation changes in #425.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

Using a machine readable file with stricter formatting requirements to store this information should reduce human error and also make it easier for third parties to ingest our data. Using separate YAML files will ensure that pull requests using the GitHub Web UI is a viable option for less technical contributors.

This new method is reliant on the changes introduced in [v1.1.0 of our Python library related to this effort](https://github.com/cisagov/log4j-md-yml/releases/tag/v1.1.0).
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass. Functionality testing was made in [this branch of a testing repository](https://github.com/cisagov/github-actions-playground/tree/testing/log4j_list_update).
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

mcdonnnj committed:
Add list updating workflow and configuration

Add the GitHub Actions workflow that will process the YAML files that
contain cisagov controlled software information and generate a final
Markdown file. The required template and Python requirements are
included as well.

mcdonnnj committed:
Update the yamllint rules

Controlling the line length in the files generated from the Markdown to
YAML conversion would be difficult so we disable that yamllint rule.

mcdonnnj committed:
Remove product status from issue forms

This value is derived from other information in each software product
entry in the YAML file. Since it is now a derived value we should not
prompt submitters to provide one.

mcdonnnj committed:
Generate YAML files from the existing Markdown

Generate the individual YAML files that are to be edited and then
generate the initial comprehensive YAML file that is used to generate
the Markdown file for display.

Lcerkov reviewed (approved):
Everything looks as expected!

dav3r commented on config/SOFTWARE-LIST.tpl.md:
I think we should mention something like this:
```suggestion
NOTE: This file is automatically generated.  To make updates, please refer to
[`CONTRIBUTING.md`](CONTRIBUTING.md).

{{software_markdown_table}}
```

Lcerkov replied:
@dav3r Good addition!

mcdonnnj replied:
Added in 3991f5e.

dav3r reviewed (approved):
Looks great to me, though I did make one suggestion that I think should be included. 🦖

mcdonnnj committed:
Add a note to the software list template

Add a note that directs viewers of the generated SOFTWARE-LIST.md to
refer to CONTRIBUTING.md for directions on updating product
information.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj committed:
Merge branch 'develop' into improvement/switch_to_yaml_files
